### PR TITLE
[testing] compress agency dumps (#18083)

### PR DESCRIPTION
**Pull request by dothebart**
backport of https://github.com/arangodb/arangodb/pull/18083

### CODE REVIEW

File: js/client/modules/@arangodb/testutils/instance-manager.js [hunk 1]
```diff
@@ -524,20 +527,48 @@ class instanceManager {
...
+    if (fs.isFile(zipfn)) {
+      fs.remove(zipfn);
+    };
+    if (fs.exists(dumpdir)) {
+      fs.list(dumpdir).forEach(file => {
+        const fn = fs.join(dumpdir, file);
+        if (fs.isFile(fn)) {
+          fs.remove(fn);
+        }
+      });
```
Comment (alexbakharew) [inline js/client/modules/@arangodb/testutils/instance-manager.js:541]: Why we can't just remove all dumpdir without iterating over it? 
Why we delete here only files but not directories?

May be we can use something like this?
```
fs.rmSync(dir, { recursive: true, force: true });
```

File: js/client/modules/@arangodb/testutils/instance-manager.js [hunk 2]
```diff
@@ -524,20 +527,48 @@ class instanceManager {
...
         }
       }
     });
+    let zipfiles = [];
+    fs.list(dumpdir).forEach(file => {
+      const fn = fs.join(dumpdir, file);
+      if (fs.isFile(fn)) {
+        zipfiles.push(file);
+      }
+    });
```
Comment (alexbakharew) [inline js/client/modules/@arangodb/testutils/instance-manager.js:565]: Same question here: Why we add to zip only files but not directories and content of them?

**Review by alexbakharew — CHANGES_REQUESTED** [review]
Need to figure out some questions

File: js/client/modules/@arangodb/testutils/instance-manager.js (hunk 1 above)
Comment (dothebart) [reply js/client/modules/@arangodb/testutils/instance-manager.js:541]: there may be zips and other json reports in this directory,

File: js/client/modules/@arangodb/testutils/instance-manager.js (hunk 2 above)
Comment (dothebart) [reply js/client/modules/@arangodb/testutils/instance-manager.js:565]: `..` is a directory ;)

**Review by naushniki — APPROVED** [review]
LGTM.

File: js/client/modules/@arangodb/testutils/instance-manager.js (hunk 1 above)
Comment (alexbakharew) [reply js/client/modules/@arangodb/testutils/instance-manager.js:541]: So we will delete only files on the first leve of `dumpdir`, right? Directories will be not processed?

**Review by alexbakharew — APPROVED** [review]
LGTM

**Review by KVS85 — APPROVED** [review]

**Pull request merged**